docs/api/synchronization.rst

```diff
@@ -0,0 +1,160 @@
+with the *Kinto* server. Remote changes are downloaded and applied
+on the local data. Local changes are uploaded using HTTP headers to control
+concurrency and overwrite.
+
+In short:
+
+#. Poll for remote changes using ``?_since=<timestamp>``
+#. Apply changes locally
+#. Post creations
+#. Use concurrency control to send updates and deletes
+
+
+Polling for changes
+===================
+
+When fetching the collection records with ``?_since=<timestamp>``, *Kinto* returns
+every records that were created/updated/deleted **after** this timestamp.
+
+#. Cold sync → no ``?_since``.
+#. ``GET /buckets/default/collections/articles?_since=<timestamp>`` (with header ``If-Unmodified-Since = <timestamp>``)
+#. If response is ``304 Not modified``, done → up to date, nothing to do.
+#. If response is ``200 OK`` store ``ETag`` response header value for next
+   synchronization and handle the obtained list of changes.
+
+.. note::
+
+    * Deleted records have an attribute ``delete: true``
+    * Created/updated records are both returned in their new version
+    * Since *Kinto* does not keep any history, there is no *diff* for updates
+
+
+Paginated changes
+-----------------
+
+By default, *Kinto* does not paginate the records list. If an explicit limit is
+set in the server settings or using the ``?_limit=<nb>`` parameter, then polling for
+changes will be paginated.
+
+It basically consists in fetching the list until the ``Next-Page`` is present.
+
+Since changes can occur between the first and the last page, the synchronization
+process is a bit more complex.
+
+#. Cold sync → no ``?_since``.
+#. ``GET /buckets/default/collections/articles?_since=<timestamp>`` (with header ``If-Unmodified-Since = <timestamp>``)
+#. If response is ``304 Not modified``, done → up to date, nothing to do.
+#. If no ``Next-Page`` response header, done → store ``ETag`` response header valuè for next
+   synchronization.
+#. If ``Next-Page`` response header is present → store the ``ETag`` response header value into a variable ``timestamp``
+   and go on to it (it's an url) using a ``If-Match: <timestamp>`` request header.
+#. If response is ``200 OK`` → repeat previous step.
+#. If response is ``412 Precondition Failed``, some changes occured since the last
+   page → Store the ``ETag``response header into ``before``
+#. Fetch and handle the changes using ``GET collections?_since=<timestamp>&_before=<before>``,
+   and store the ``ETag``response header in to ``timestamp``
+#. Go back to step 5 (follow the ``Next-Page``)
+
+
+Apply changes
+=============
+
+Applying remote changes to the local database consists in adding new records,
+updating changed records and remove deleted records.
+
+From the client perspective, *Kinto* does not distinguish creations from updates.
+In the *polling for changes* response, created records are simply the records
+unknown by the client (using ``id`` field).
+
+If the records to be updated or deleted had also been modified locally then
```

Comment:
> In the case that the record marked for update or deletion has also been modified locally, it will be up to the developer to choose a strategy - for example, *merge fields* or *ignore deletion*.